File: _includes/code/python/local.quickstart.query.rag.py
import time

time.sleep(
    5
)  # Allow Weaviate data import to finish TODO[g-despot]: Implement a better solution to wait for data import to finish
# RAG
import weaviate
from weaviate.classes.generate import GenerativeConfig

client = weaviate.connect_to_local()

questions = client.collections.use("Question")

# highlight-start
response = questions.generate.near_text(
    query="biology",
    limit=2,
    grouped_task="Write a tweet with emojis about these facts.",
    generative_provider=GenerativeConfig.ollama(  # Configure the Ollama generative integration
        api_endpoint="http://ollama:11434",  # If using Docker you might need: http://host.docker.internal:11434
        model="llama3.2",  # The model to use
    ),
)
# highlight-end

print(response.generative.text)  # Inspect the generated text

client.close()  # Free up resources
# END RAG

File: _includes/code/typescript/local.quickstart.query.rag.ts
// RAG
import weaviate, { WeaviateClient, generativeParameters } from 'weaviate-client';

const client: WeaviateClient = await weaviate.connectToLocal();

// highlight-start
const questions = client.collections.use('Question');

const result = await questions.generate.nearText(
  'biology',
  {
    groupedTask: 'Write a tweet with emojis about these facts.',
    config: generativeParameters.ollama({
      apiEndpoint: 'http://ollama:11434',  // If using Docker you might need: http://host.docker.internal:11434
      model: 'llama3.2',                   // The model to use
    }),
  },
  {
    limit: 2,
  }
);
// highlight-end

console.log(result.generative);

client.close(); // Close the client connection
// END RAG
